### DISCUSSION TITLE
devServer Not effective

### DISCUSSION THREAD

**Discussion by ayoullf** [webpack/webpack#15429] [Q&A]
# Bug report

**What is the current behavior?**
i want to config the proxy,but it doesn't effective,request report error 404.my config:
          output: {
		          path: path.join(__dirname, "./dist"),
		          publicPath: "/dist/",
		          filename: "bundle.js"
	          },
	devServer: {
		proxy: {
			'/dabuApi': {                
				target: 'http://127.0.0.1:1688',         
				changeOrigin: true, 
				pathRewrite:{'^/dabuApi':''},
			},
		},
	},

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
![image](https://user-images.githubusercontent.com/32232583/155070570-41ad7244-4c66-48b0-ab27-f2e2ee8ec084.png)

**Other relevant information:**
webpack version:  2.2
Node.js version: v14.17.5
Operating System: 
Additional tools:

**Comment by alexander-akait** [comment]
Please fill out all fields

**Reply by alexander-akait** [reply]
And check your publicPath

**Reply by ayoullf** [reply]
here's all fields 
var path = require("path");
var webpack = require('webpack');

module.exports = {
	entry: "./src/index.js",
    devtool: "#cheap-module-eval-source-map",
	output: {
		path: path.join(__dirname, "./dist"),
		publicPath: "/dist/",
		filename: "bundle.js"
	},
	devServer: {
		proxy: {
			'/dabuApi': {             
				target: 'http://127.0.0.1:1688',        
				changeOrigin: true, 
				pathRewrite:{'^/dabuApi':'/'},
			},
		},
	},
	module: {
		loaders: [
			{test:/\.vue$/, loader: 'vue-loader'},
			{test:/\.css$/, loader: 'style-loader!css-loader'},
			{	
				test: /\.(jpe?g|png|gif|svg)$/,
                loader: 'url-loader?limit=1024',
            },
            {
            	test: /\.(eot|svg|ttf|woff|woff2)(\?\S*)?$/,
            	loader: "file-loader"
            },
            {
            	test: /\.js/, 
            	loader: "babel-loader",
            	exclude: "/node_modules/"
        	},
        	{
        		test: /\.less/,
        		loader: "style-loader!css-loader!less-loader!"
        	}
		]
	},
	resolve: {
        alias: {
            'vue': 'vue/dist/vue.js'
        }
    }
}

if(process.env.NODE_ENV === 'production') {
    module.exports.devtool = '#cheap-module-source-map';
    module.exports.plugins = (module.exports.plugins || []).concat([
        new webpack.DefinePlugin({
            'process.env': {
                NODE_ENV: '"production"'
            }
        }),
        new webpack.optimize.UglifyJsPlugin({
            compress: {
                warnings: false
            }
        })
    ])
}

**Reply by alexander-akait** [reply]
Can you provide reproducible example? Looks like `dabuApi` is not applied

**Reply by ayoullf** [reply]
sorry, I can't catch your meaning. 'reproducible example' mean?@V@

**Reply by ayoullf** [reply]
i try to upgrade my webpack's version but not success.my project has too many low-version dependencies

**Reply by alexander-akait** [reply]
I mean can you provide exampl eof the problem on https://github.com/, i.e. create small repo, it is hard to say what is wrong without versions and code

**Reply by 865881900** [reply]
'^/dabuApi'   => /dabuApi 
Give it a try
